repositorycreation/create_repository.go

```diff
@@ -0,0 +1,221 @@
+	if err != nil {
+		log.Event(ctx, "update branch protection failed for develop", log.Error(err))
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "master")
+	if err != nil {
+		log.Event(ctx, "adding protection, require signatures failed on branch master", log.Error(err), log.Data{"response": resp})
+	}
+	_, resp, err = client.Repositories.RequireSignaturesOnProtectedBranch(ctx, org, repoName, "develop")
+	if err != nil {
+		log.Event(ctx, "adding protection, require signatures failed on branch develop", log.Error(err), log.Data{"response": resp})
+	}
+	return err
+}
+
+// setDevelopAsDefaultBranch will set the develop branch as the repositories default branch
+func setDevelopAsDefaultBranch(ctx context.Context, client *github.Client, repoName string, repo *github.Repository) error {
+	repo.AutoInit = nil
+	_, resp, err := client.Repositories.Edit(ctx, org, repoName, repo)
+	if err != nil {
+		log.Event(ctx, "failed to set develop as the default branch", log.Error(err), log.Data{"response": resp})
+	}
+	return err
+}
+
+// createDevelopBranch will create a new branch named /develop, based on the last ref on master branch
+func createDevelopBranch(ctx context.Context, client *github.Client, repoName string) error {
+	ref, resp, err := client.Git.GetRef(ctx, org, repoName, "heads/master")
+	if err != nil {
+		log.Event(ctx, "get reference to master commit failed", log.Error(err), log.Data{"response": resp})
+		return err
+	}
+	developBranch := "heads/develop"
+	ref.Ref = &developBranch
+
+	_, resp, err = client.Git.CreateRef(ctx, org, repoName, ref)
+	if err != nil {
+		log.Event(ctx, "create Reference to new develop branch failed", log.Error(err), log.Data{"response": resp})
+		return err
+	}
+	return nil
+}
+
+// createRepo makes a call via the gitHubAPI to generate a new repository
+func createRepo(ctx context.Context, client *github.Client, repo *github.Repository) error {
+	newRepo, _, err := client.Repositories.Create(ctx, org, repo)
+	fmt.Println(newRepo)
+	if err != nil {
+		log.Event(ctx, "repo creation failed", log.Error(err))
+		return err
+	}
+	return nil
+}
+
+// getConfigurationsForNewRepo gets required configuration information from the end user
+func getConfigurationsForNewRepo() (accessToken, userHandle, repoName, repoDescription, defaultBranch string) {
+	accessToken = promptForInput("Please provide your user access token, to create one follow this guide https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line")
+	userHandle = promptForInput("Please provide your github handle/username")
+	repoName = promptForInput("Please provide the full name for the new repository")
+	repoDescription = promptForInput("Please provide a description for the repository")
+	defaultBranch = promptForInput("Please pick a default branch, either: 'develop' or 'master'")
```

Comment:
Rather than asking for the default branch it might be worth asking for the branching strategy (i.e. `git flow` vs `github flow`).  I would also suggest that for this type of input it is advisable to offer numbered options:

```
Please pick the branching strategy you wish this repo to use:

[1] github flow
[2] git flow

Please enter the number corresponding to your choice:
```

Reply:
So the idea of asking for the default branch here was that on libraries the default branch is 'master' whereas runnable apps have a default branch of 'master'.  So not sure it is really a quesion of which work flow to use. I can / will switch over to using that style prompt mind.

Reply:
Hi Dan, I know I was the one that suggested to split the branching strategy based on whether the repo is for a library or application but it would make more sense to base it on the type of branching strategy as James put it.

The end results are the same for what we need to do but abstracting it back to the branching strategy would be better as the repo maybe used for different reasons other than a library or application.